tests/conftest.py

```
@@ -0,0 +1,262 @@
+    gc.collect()
+
+    # Log memory statistics if profiling enabled
+    try:
+        import os
+
+        if os.environ.get("PYTEST_MEMORY_PROFILE", "0") == "1":
+            import psutil
+
+            process = psutil.Process()
+            memory_info = process.memory_info()
+            print(f"\n=== Test Suite Memory Statistics ===")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)